packages/model-viewer/src/three-components/gltf-instance/ModelViewerGLTFInstance.ts

```diff
@@ -185,21 +180,51 @@ export class ModelViewerGLTFInstance extends GLTFInstance {
       clone.normalMap = material.normalMap.clone();
       clone.normalMap.needsUpdate = true;
     }
-    if (material.aoMap != null) {
-      clone.aoMap = material.aoMap.clone();
-      clone.aoMap.needsUpdate = true;
-    }
     if (material.emissiveMap != null) {
       clone.emissiveMap = material.emissiveMap.clone();
       clone.emissiveMap.needsUpdate = true;
     }
-    if (material.metalnessMap != null) {
+
+    // Clones the roughnessMap if it exists.
+    let roughnessMap: Texture|null = null;
+    if (material.roughnessMap) {
+      roughnessMap = material.roughnessMap!.clone();
+    }
+
+    // Assigns the roughnessMap to the cloned material and generates mipmaps.
+    if (roughnessMap !== null) {
+      roughnessMap.needsUpdate = true;
+      clone.roughnessMap = roughnessMap;
+
+      // Generates mipmaps from the clone of the roughnessMap
+      // TODO (b/188906197) Render artifacts when generating mipmaps after
```

Comment:
Let's avoid internal references since this code is all public. Let's put a three.js bug link here instead if we have one.

Reply:
Done.